lib/form_data.js

```diff
@@ -192,7 +210,8 @@ FormData.prototype._getContentDisposition = function(value, options) {
 
   // custom filename takes precedence
   // fs- and request- streams have path property
-  var filename = options.filename || value.path;
+  // formidable and the browser add a name property.
+  var filename = options.filename || value.name || value.path;
```

Comment:
This is the one that is related to #163. The reason I added it is that in the browser the [`name`](https://developer.mozilla.org/en-US/docs/Web/API/File/name) property of a file object is what is used to determine the filename. Formidable's file also has a [`name`](https://github.com/felixge/node-formidable#user-content-formidablefile) property which could be used if the file was a read stream (typically formidable files are only write streams but in (one of my libs)[https://github.com/felixge/node-formidable#user-content-formidablefile
https://github.com/rill-js/body/blob/master/server/index.js#L72] I retrofit it to be a read/write stream. Also some isomorphic libraries such as (Node-File-API)[user-content-file
https://github.com/node-file-api/file-api#user-content-file] implement the same interface so it's mainly just a convenience that allows for more ways to infer the file name of the appended form data.

Again I can remove this part of the PR if it is the make/break.